beacon-chain/state/stategen/setter.go

```diff
@@ -81,3 +96,44 @@ func (s *State) saveStateByRoot(ctx context.Context, blockRoot [32]byte, state *
 
 	return nil
 }
+
+// EnableSaveHotStateToDB enters the mode that saves hot beacon state to the DB.
+// This usually gets triggered when there's long duration since finality.
+func (s *State) EnableSaveHotStateToDB(_ context.Context) {
+	s.saveHotStateDB.lock.Lock()
+	defer s.saveHotStateDB.lock.Unlock()
+	if s.saveHotStateDB.enabled {
+		return
+	}
+
+	s.saveHotStateDB.enabled = true
+
+	log.WithFields(logrus.Fields{
+		"enabled":       s.saveHotStateDB.enabled,
+		"slotsInterval": s.saveHotStateDB.duration,
+	}).Warn("Entering mode to save hot states in DB")
+}
+
+// DisableSaveHotStateToDB exits the mode that saves beacon state to DB for the hot states.
+// This usually gets triggered once there's finality after long duration since finality.
+func (s *State) DisableSaveHotStateToDB(ctx context.Context) error {
+	s.saveHotStateDB.lock.Lock()
+	defer s.saveHotStateDB.lock.Unlock()
+	if !s.saveHotStateDB.enabled {
+		return nil
+	}
+
+	// Delete previous saved states in DB as we are turning this mode off.
+	if err := s.beaconDB.DeleteStates(ctx, s.saveHotStateDB.savedStateRoots); err != nil {
+		return err
+	}
+	s.saveHotStateDB.enabled = false
```

Comment:
Why is it important to only disable this if DeleteStates doesn't fail?

Reply:
You are right. I think we should disable even if `DeleteStates` fails.

Reply:
It is OK to do it this way, my request is just some clarification. Maybe in a comment. 

Why would DeleteStates fail? 
Could it end up in a loop where we are frequently trying to disable save hot state DB but can't for some reason?

Reply:
It's hard to come up with an argument on why `DeleteStates` could fail here. Maybe times out?

With that said, the priority should be given to `s.saveHotStateDB.enabled = false`

Reply:
I think that is the safe option. Thanks